gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/GfxdSystemProcedures.java

```diff
@@ -2568,8 +2563,8 @@ public static String GET_SNAPSHOT_TXID() throws SQLException, StandardException
     }
     // tc commit will clear all the artifacts but will not commit actual txState
     // that should be committed in COMMIT procedure
-    tc.resetActiveTXState(true);
-    TXManagerImpl.getOrCreateTXContext().clearTXState();
+    tc.clearActiveTXState(true, true);
+    //tc.resetActiveTXState(true);
```

Comment:
@sumwale Initially did use ThreadLocal connection, however there was an issue with colocated join where two connections were required for two different tables.
Second approach can be used for atleast single conn for column table iteration without affecting joins. Will try that.